tests/ExceptionTest.php

```diff
@@ -57,9 +51,17 @@ public function testNonexistantFieldException()
     {
         $c = Connection::connect('sqlite::memory:');
         $q = $c->dsql();
-        $q->table('foo')->field('do_not_exist');
+        $q->table('dual')->field('do_not_exist');
+
+        $this->setExpectedException(\atk4\dsql\ExecuteException::class);
 
-        $this->setExpectedException('atk4\dsql\ExecuteException');
-        $q->execute(); // PDOException: SQLSTATE[HY000]: General error: 1 no such table: foo
+        try {
+            $q->execute();
+        } catch (\atk4\dsql\ExecuteException $e) {
+            $this->assertSame(1054, $e->getCode());
```

Comment:
This is my idea to test, but still do not know how to query MySQL/MariaDB here

Reply:
How about using `driver` property? It's `sqlite`, `mysql` or `pgsql` or `oracle` - taken from Persistence class.

Reply:
Why not, but I can not really open hundreds PRs for these smaller fixes.